ai-service/app/routers/price.py

```
@@ -0,0 +1,56 @@
+def get_predictor() -> PricePredictor:
+    """Get or initialize the price predictor singleton."""
+    global _predictor
+    if _predictor is None:
+        _predictor = PricePredictor()
+        _predictor.load_model()
+    return _predictor
+
+@router.post("/predict", response_model=PricePredictionResponse)
+async def predict_price(
+    request: PricePredictionRequest,
+    predictor: PricePredictor = Depends(get_predictor),
```

Ruff (0.16.2)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)